### PR TITLE
Mention that displaySurface cannot be changed in JS

### LINKED ISSUE #1456: MediaTrackConstraints.displaySurface: there is no way to restrain surface type via javascript; can only detect surface type for a current stream already going
@simajzenman commented on [Tue Nov 10 2020](https://github.com/mdn/sprints/issues/3878)

# Request type

- [ ] Correction or update

# Details

Hello Mozilla,

I've seen in the page about MediaTrackConstraints.displaySurface (https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints/displaySurface) that in the first paragraph, it says "is used to specify the type or types of display surfaces which getDisplayMedia() will let the user select among for sharing purposes.". As far as I know (and tested) there is no way to restrain the type of surface you want to allow in javascript on client side; displaySurface (as it is written in the same page) can be used to detect the type of surface for a current stream already going.

If I am wrong, I first apologies for wasting your time and would be happy to have more info.
I also tried to edit it myself but couldn't for that paragraph

In the meantime, if you know where I could find in the doc a way to identify the stream name(meaning if I share facebook tab browser, I should find "facebook" in the name of the current stream), it would really help me.

Have a great one,
Simon

### PR DESCRIPTION
**Pull request by dscotese**
### Description

I reviewed https://w3c.github.io/mediacapture-screen-share/#dfn-displaysurface and did the best I could.  The spec "document is not complete. It is subject to major changes and, while early experimentations are encouraged, it is therefore not intended for implementation."

### Motivation

The issue is old and the spec it addresses is still incomplete, but since the issue was open, I took a look and believe my changes will help to explain how the displaySurface property can be used to enhance screen sharing through browsers.

### Additional details

See https://w3c.github.io/mediacapture-screen-share/#dfn-displaysurface as mentioned above.

### Related issues and pull requests

This might fix #1456

### CODE REVIEW

**Comment by dscotese** [comment]
I don't see the problem here with "Markdownlint (PR files) / lint-docs (pull_request)".  Can anyone explain it to me so I can fix it?

File: files/en-us/web/api/mediatrackconstraints/displaysurface/index.md [hunk 1]
```diff
@@ -8,16 +8,16 @@ browser-compat: api.MediaTrackConstraints.displaySurface
 
 {{APIRef("Media Capture and Streams")}}
 
-The {{domxref("MediaTrackConstraints")}} dictionary's **`displaySurface`** property is a [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) describing the requested or mandatory constraints placed upon the value of the {{domxref("MediaTrackSettings.displaySurface","displaySurface")}} constrainable property.
+The {{domxref("MediaTrackConstraints")}} dictionary's **`displaySurface`** property is a [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) describing the preferred value for the {{domxref("MediaTrackSettings.displaySurface","displaySurface")}} constrainable property.
 
-This is used to specify the type or types of display surfaces which {{domxref("MediaDevices.getDisplayMedia", "getDisplayMedia()")}} will let the user select among for sharing purposes.
+This is set by the application to identify to the user agent the type of display surface (window, browser, or monitor at this time) preferred by the application. It has no effect on what the user can choose to share, but may be used to present the options in a different order.
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediatrackconstraints/displaysurface/index.md:13]:
```suggestion
This is set by the application to identify to the user agent the type of display surface (window, browser, or monitor) preferred by the application. It has no effect on what the user can choose to share, but may be used to present the options in a different order.
```

File: files/en-us/web/api/mediatrackconstraints/displaysurface/index.md [hunk 2]
```diff
@@ -8,16 +8,16 @@ browser-compat: api.MediaTrackConstraints.displaySurface
...
 
 If needed, you can determine whether or not this constraint is supported by checking the value of {{domxref("MediaTrackSupportedConstraints.displaySurface")}} as returned by a call to {{domxref("MediaDevices.getSupportedConstraints()")}}. However, typically this is unnecessary since browsers will ignore any constraints they're unfamiliar with.
 
 ## Value
 
-A [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) which specifies the type of display surface that's being captured.
-This value _does not_ affect the list of display sources in the browser's user interface.
+A [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) which specifies the type of display surface preferred by the application.
+This value _does not_ add or remove display sources in the browser's user interface, but may reorder them. You can't use this property to limit the user to a subset of the three display surface values, window, browser, and monitor, but, as you will see below, you can see what was chosen and reject it.
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediatrackconstraints/displaysurface/index.md:20]:
```suggestion
This value _does not_ add or remove display sources in the browser's user interface, but may reorder them. You can't use this property to limit the user to a subset of the three display-surface values `window`, `browser`, and `monitor` — but, as you will see below, you can see what was chosen, and reject it.
```

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/mediatrackconstraints/displaysurface/index.md [hunk 3]
```diff
@@ -8,16 +8,16 @@ browser-compat: api.MediaTrackConstraints.displaySurface
 
 {{APIRef("Media Capture and Streams")}}
 
-The {{domxref("MediaTrackConstraints")}} dictionary's **`displaySurface`** property is a [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) describing the requested or mandatory constraints placed upon the value of the {{domxref("MediaTrackSettings.displaySurface","displaySurface")}} constrainable property.
+The {{domxref("MediaTrackConstraints")}} dictionary's **`displaySurface`** property is a [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) describing the preferred value for the {{domxref("MediaTrackSettings.displaySurface","displaySurface")}} constrainable property.
 
-This is used to specify the type or types of display surfaces which {{domxref("MediaDevices.getDisplayMedia", "getDisplayMedia()")}} will let the user select among for sharing purposes.
+This is set by the application to identify to the user agent the type of display surface (window, browser, or monitor) preferred by the application. It has no effect on what the user can choose to share, but may be used to present the options in a different order.
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediatrackconstraints/displaysurface/index.md:13]:
```suggestion
This is set by the application to identify to the user agent the type of display surface (`window`, `browser`, or `monitor`) preferred by the application. It has no effect on what the user can choose to share, but may be used to present the options in a different order.
```

File: files/en-us/web/api/mediatrackconstraints/displaysurface/index.md [hunk 4]
```diff
@@ -8,16 +8,16 @@ browser-compat: api.MediaTrackConstraints.displaySurface
...
 
 If needed, you can determine whether or not this constraint is supported by checking the value of {{domxref("MediaTrackSupportedConstraints.displaySurface")}} as returned by a call to {{domxref("MediaDevices.getSupportedConstraints()")}}. However, typically this is unnecessary since browsers will ignore any constraints they're unfamiliar with.
 
 ## Value
 
-A [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) which specifies the type of display surface that's being captured.
-This value _does not_ affect the list of display sources in the browser's user interface.
+A [`ConstrainDOMString`](/en-US/docs/Web/API/MediaTrackConstraints#constraindomstring) which specifies the type of display surface preferred by the application.
+This value _does not_ add or remove display sources in the browser's user interface, but may reorder them. You can't use this property to limit the user to a subset of the three display-surface values `window`, `browser`, and `monitor` — but, as you will see below, you can see what was chosen, and reject it.
```
Comment (sideshowbarker) [inline files/en-us/web/api/mediatrackconstraints/displaysurface/index.md:20]:
```suggestion
This value _does not_ add or remove display sources in the browser's user interface, but may reorder them. You can't use this property to limit the user to a subset of the three display-surface values `window`, `browser`, and `monitor` — but, as you will see below, you can see what was chosen, and reject it.
```

**Comment by sideshowbarker** [comment]
> I don't see the problem here with "Markdownlint (PR files) / lint-docs (pull_request)". Can anyone explain it to me so I can fix it?

It was just this:

image

In other words, Prettier requires double quotes there, rather than single quotes.

But from the error message our check here is reporting, there’s no way you could have known that was the cause.

**Comment by dscotese** [comment]
> It was just this: ...

How did you figure it out, and is there a project I can work on to see if I can make the error more informative?

...and Thanks!

**Comment by sideshowbarker** [comment]
> > It was just this: ...
> 
> How did you figure it out

I ran prettier on the file from the command line with the `-w` (write) switch, and then looked at the diff.

> and is there a project I can work on to see if I can make the error more informative?

I think the source for the linter is in the mdn/yari repo, so you’d want to look somewhere in there.

But I’d suggest first raising an issue with a pointer to this PR, and description like _“Linter (or Prettier) error message doesn’t report what the actual problem is (e.g., single quotes used rather than double quotes).”_

**Pull request merged**